rasa/telemetry.py

```diff
@@ -484,6 +509,56 @@ def toggle_telemetry_reporting(is_enabled: bool) -> None:
     rasa_utils.write_global_config_value(CONFIG_FILE_TELEMETRY_KEY, configuration)
 
 
+@ensure_telemetry_enabled
+def initialize_error_reporting() -> None:
+    """Sets up automated error reporting.
+
+    Exceptions are reported to sentry. We avoid sending any metadata (local
+    variables, paths, ...) to make sure we don't compromise any data. Only the
+    exception and its stacktrace is logged and only if the exception origins
+    from the `rasa` package. """
+    import sentry_sdk
+    from sentry_sdk.integrations.atexit import AtexitIntegration
+    from sentry_sdk.integrations.dedupe import DedupeIntegration
+    from sentry_sdk.integrations.excepthook import ExcepthookIntegration
+
+    # key for local testing can be found at
+    # https://sentry.io/settings/rasahq/projects/rasa-open-source/install/python/
+    # for local testing, set the key using `RASA_EXCEPTION_WRITE_KEY=key rasa <command>`
+    key = sentry_write_key()
+
+    if not key:
+        return
+
+    def strip_sensitive_data(event, hint):
+        # removes any paths from stack traces (avoids e.g. sending
+        # a users home directory name if package is installed there)
+        for value in event.get("exception", {}).get("values", []):
+            for frame in value.get("stacktrace", {}).get("frames", []):
+                frame["abs_path"] = ""
+        return event
+
+    # this is a very defensive configuration, avoiding as many integrations as
+    # possible. it also submits very little data (exception with error message
+    # and line numbers).
+    sentry_sdk.init(
+        f"https://{key}.ingest.sentry.io/2801673",
+        before_send=strip_sensitive_data,
+        integrations=[
+            ExcepthookIntegration(),
```

Comment:
optional: how about using https://docs.sentry.io/platforms/python/guides/logging/ + changing the sanic error logging to `logger.exception` so that we can also catch server errors?

Reply:
I took a closer look at that and even though I think it is something we should do in the future it is to much effort to do it now: we can't just hook into the exception handling, as sanic uses exceptions to bubble up any errors (even expected ones) which would create to much noise. so this means this needs a bit more work to filter out any exceptions that are expected to reduce the logging noise. let's do that after the RC

Reply:
Can't we hook into https://github.com/RasaHQ/rasa/blob/f1055d18b46071312259057a23c80ce5506e158f/rasa/server.py#L90 and only do it for `5xx` status codes? Or do you think that's still too much noise?

Reply:
it might work, but I think it requires a bit more testing that I'd like to avoid just now given the current timeline. I think it's something that we can easily add after this is merged